### PR TITLE
Support for top level platform declarations

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: test/goldens/end2end/audio_service-0.17.0.json_report.md [hunk 1]
```diff
@@ -15,23 +15,70 @@
...
 * Found example at: `example/lib/main.dart`
 
-## 20/20 Support multiple platforms
+## 20/20 Platform Support
 
-### [*] 20/20 points: Supports 3 of 3 possible platforms (**iOS**, **Android**, **Web**)
+### [*] 20/20 points: Supports 4 of 6 possible platforms (**iOS**, **Android**, **Web**, Windows, **MacOS**, Linux)
 
-Found 8 issues. Showing the first 2:
+Found 12 issues. Showing the first 10:
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply test/goldens/end2end/audio_service-0.17.0.json_report.md]: I fixed the issue-count to only count "Issues".

**Comment by sigurdm** [comment]
> At what point do we want to remove the runtime: tags?

I think they still make a lot of sense - you might be interested in packages that work on dart-native, not caring about the actual platform....

File: test/goldens/end2end/dnd-2.0.0.json_report.md [hunk 2]
```diff
@@ -15,16 +15,53 @@
...
 
+* ✓ Web
 
-Consider supporting multiple platforms:
+These platforms are not supported:
 
 <details>
 <summary>
-Package not compatible with runtime native
+Package not compatible with runtime native on Android
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply test/goldens/end2end/dnd-2.0.0.json_report.md]: Good point - Done

**Review by isoos — APPROVED** [review]
lgtm

**Review by jonasfj — APPROVED** [review]
LGTM; I guess we could remove the `runtime:...` tags once the new search UI has shipped, right?

**Pull request merged**